### ISSUE TITLE
Emit some information about what target the Docker extension is trying to connect to

### ISSUE THREAD

**Issue by webdog** [microsoft/vscode-docker#2493]
Version: 1.51.0
Commit: fcac248b077b55bae4ba5bab613fd6e9156c2f0c
Date: 2020-11-05T18:14:40.758Z (1 wk ago)
Electron: 9.3.3
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 19.4.0

- VSCode Version: 1.51.0
- Local OS Version: 10.15.4
- Remote OS Version: N/A
- Remote Extension/Connection Type: SSH/Docker/WSL Docker

Steps to Reproduce:

1. Edit docker.host to tcp://`$docker_host_ip`:2376
2. Refresh Containers

Does this issue occur when you try this locally?: Yes/No Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes/No Yes

I am setting `docker.host` to connect to my local installation of Docker for Desktop on Mac, which works and can connect with no issue independently of VSCode. However VSCode complains that it cannot verify the first certificate:
![image](https://user-images.githubusercontent.com/960210/98989895-5b270680-24ef-11eb-92d9-93e540a07a55.png)
![image](https://user-images.githubusercontent.com/960210/98988775-c079f800-24ed-11eb-82de-66555803568a.png)
![image](https://user-images.githubusercontent.com/960210/98989034-24042580-24ee-11eb-824a-9c7a8541182b.png)
![image](https://user-images.githubusercontent.com/960210/98990533-32ebd780-24f0-11eb-8e02-0ed497d8d462.png)

```sh
(base) ➜  echo $PATH
/Users/christianweber/opt/anaconda3/bin:/Users/christianweber/opt/anaconda3/condabin:/opt/anaconda3/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
(base) ➜  docker-machine env
export DOCKER_TLS_VERIFY="1"
export DOCKER_HOST="tcp://192.168.99.100:2376"
export DOCKER_CERT_PATH="/Users/christianweber/.docker/machine/machines/default"
export DOCKER_MACHINE_NAME="default"
# Run this command to configure your shell: 
# eval $(docker-machine env)

(base) ➜  env (output edited for brevity)
SHELL=/bin/zsh
DOCKER_TLS_VERIFY=1
DOCKER_HOST=tcp://192.168.99.100:2376
DOCKER_CERT_PATH=/Users/christianweber/.docker/machine/machines/default
DOCKER_MACHINE_NAME=default

(base) ➜  docker-machine ip
192.168.99.100

(base) ➜  ping 192.168.99.100
PING 192.168.99.100 (192.168.99.100): 56 data bytes
64 bytes from 192.168.99.100: icmp_seq=0 ttl=64 time=0.314 ms
64 bytes from 192.168.99.100: icmp_seq=1 ttl=64 time=0.268 ms

(base) ➜  docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                                      NAMES
b4ecef67142b        webdog              "/docker-entrypoint.…"   2 hours ago         Up 2 hours          80/tcp, 8000/tcp, 0.0.0.0:8080->8080/tcp   peaceful_hamilton
```
Console logging does not provide any debug logging when displaying the front-end certificate error:
![image](https://user-images.githubusercontent.com/960210/98989389-a856a880-24ee-11eb-8358-289f5ba9b308.png)

Additionally, trying to enter just the IP address of the local host results in a formatting error:
![image](https://user-images.githubusercontent.com/960210/98989553-e94ebd00-24ee-11eb-8c0c-f9304488bad4.png)
![image](https://user-images.githubusercontent.com/960210/98989597-f4a1e880-24ee-11eb-8e8d-2b8101b31871.png)
![image](https://user-images.githubusercontent.com/960210/98989625-fc618d00-24ee-11eb-8de9-5a2558c89c47.png)

I think I have two questions:
* Is it necessary to have certificate verification on a local install of Docker?
* Is the certificate check occurring because this docker_host isn't set to localhost?

**Comment by bwateratmsft** [comment]
@webdog The `docker.host` setting does permit sockets as well as addresses; for example you could do `unix:///var/run/docker.sock` (which is what is used by default). We don't that default value there because on Windows it is instead `npipe:////./pipe/docker_engine`. As far as I know there's no way for extension authors to specify different default values per OS. As you found though, we implicitly use those values as defaults; indeed it seems most people are set up that way (and the remainder are mostly SSH).

As for the invalid URL issue, Dockerode, which we're using for connecting to Docker, requires a full URL including protocol (and in the case of SSH, username as well), which is why it won't accept `1.2.3.4:2376`, but does accept `tcp://1.2.3.4:2376`.

We could emit information to the Output window about where Docker is trying to connect to, but other than that I don't think there's a better place to give such info.

**Comment by webdog** [comment]
>As far as I know there's no way for extension authors to specify different default values per OS.

That's a really good point. Even if it did, it could become confusing quickly unless those values were nested/organized in some efficient way (Something like `system.docker.config.[windows|osx|linux]` as a passing thought). I also say that without understanding the effort required to support that, which could be considerable.

>We could emit information to the Output window about where Docker is trying to connect to, but other than that I don't think there's a better place to give such info.

Information in the output window would work, I think. It would have been a breadcrumb for me that my environment was misconfigured. In my case, my DOCKER_HOST was the IP, and if I'd seen the variable was set to the value of the Socket, I may have been able to use that info to investigate further, and understand quickly that I had two Docker installations set up (oops!)

Thanks again! I'm good with closing this out as a suggestion/feature request for a little more logging 😺

**Comment by bwateratmsft** [comment]
I'll adjust the title and set some labels. Thank you for the suggestion and for the very detailed information!

**Comment by bwateratmsft** [comment]
This is now available in Docker extension version 1.9.0.
